apps/api/src/domain/models/digital-text/digital-text.entity.ts

```diff
@@ -0,0 +1,107 @@
+import { AggregateRoot } from '../../decorators';
+import { AggregateCompositeIdentifier } from '../../types/AggregateCompositeIdentifier';
+import { AggregateId } from '../../types/AggregateId';
+import { AggregateType } from '../../types/AggregateType';
+import { ResourceType } from '../../types/ResourceType';
+import { Resource } from '../resource.entity';
+import { BaseEvent } from '../shared/events/base-event.entity';
+import { CreateDigitalText } from './commands';
+
+@AggregateRoot(AggregateType.digitalText)
+@RegisterIndexScopedCommands(['CREATE_DIGITAL_TEXT'])
+export class DigitalText extends Resource {
+    readonly type = ResourceType.digitalText;
+
+    @NestedDataType(MultilingualText, {
+        label: 'title',
+        description: 'the title of the digital text',
+    })
+    readonly title: MultilingualText;
+
+    constructor(dto: DTO<DigitalText>) {
+        super({ ...dto, type: ResourceType.digitalText });
+
+        if (!dto) return;
+
+        const { title } = dto;
+
+        this.title = new MultilingualText(title);
+    }
+
+    getName(): MultilingualText {
+        return this.title.clone();
+    }
+
+    protected getResourceSpecificAvailableCommands(): string[] {
+        return [];
+    }
+
+    protected validateComplexInvariants(): InternalError[] {
+        return [];
+    }
+
+    static fromEventHistory(
+        eventStream: BaseEvent[],
+        idOfDigitalTextToCreate: AggregateId
+    ): Maybe<ResultOrError<DigitalText>> {
+        const eventsForThisDigitalText = eventStream.filter(({ payload }) =>
+            isDeepStrictEqual((payload as ICommandBase)[AGGREGATE_COMPOSITE_IDENTIFIER], {
+                type: AggregateType.digitalText,
+                id: idOfDigitalTextToCreate,
+            })
+        );
+
+        if (eventsForThisDigitalText.length === 0) return NotFound;
+
+        const [creationEvent, ...updateEvents] = eventsForThisDigitalText;
+
+        if (creationEvent.type !== `DIGITAL_TEXT_CREATED`) {
+            throw new InternalError(
+                `The first event for ${formatAggregateCompositeIdentifier({
+                    type: AggregateType.digitalText,
+                    id: idOfDigitalTextToCreate,
+                })} should have been of type DIGITAL_TEXT, but found: ${creationEvent?.type}`
+            );
+        }
+
+        const {
+            title,
+            languageCodeForTitle,
+            aggregateCompositeIdentifier: { id, type },
+        } = creationEvent.payload as CreateDigitalText;
+
+        const initialInstance = new DigitalText({
+            type,
+            id,
+            published: false,
+            title: buildMultilingualTextWithSingleItem(title, languageCodeForTitle),
+            eventHistory: [creationEvent],
+        });
+
+        const newDigitalText = updateEvents.reduce((digitalText, event) => {
+            if (isInternalError(digitalText)) return digitalText;
+
+            if (event.type === `RESOURCE_PUBLISHED`) {
```

Comment:
Oh yea- really nice catch! There are actually more generic events (e.g. access control) that we might want to apply. One technical challenge will be finding a pattern that allows us to write these handlers once but share with all aggregate classes.